src/components/sync-connected-sites.tsx

```diff
@@ -54,13 +63,35 @@ export const convertTreeToOptionsToSync = ( tree: TreeNode[] ): SyncOption[] =>
 		const wpContent = filesAndFolders.find( ( node ) => node.id === 'wp-content' )?.children || [];
 
 		wpContent.forEach( ( item ) => {
-			if ( item.checked && isSyncOption( item.id ) ) {
+			if ( ! isSyncOption( item.id ) ) {
+				return;
+			}
+
+			if ( item.checked || item.indeterminate ) {
 				optionsToSync.push( item.id );
 			}
```

Comment:
Hm, I still don't understand this point.
1. If, for example, `plugins` item is selected, it means that all children are selected and we need just: push the `id` to `optionsToSync`, `return` and go with this flow https://github.com/Automattic/studio/pull/1495/files#diff-00ad6cb3b7847f2a153b53c275d15f0d2991f5a336d2f057cbff1af5adc0141dR189

```suggestion
			if ( item.checked ) {
				optionsToSync.push( item.id );
				return;
			}
```

2. If, for example, `plugins` item is not selected, then we should check - if at least one plugin is selected and add selected plugins to `specificSelections`. And, we don't need to push id `plugins` to `optionsToSync`, since [here](https://github.com/Automattic/studio/pull/1495/files#diff-00ad6cb3b7847f2a153b53c275d15f0d2991f5a336d2f057cbff1af5adc0141dR169) we don't use it